### DISCUSSION TITLE
Issues having two connectors talk to each other on a local machine

### DISCUSSION THREAD

**Discussion by Vincent-UBT** [International-Data-Spaces-Association/DataspaceConnector#51] [Q&A] [answered]
Hi everyone!

After some time of testing the Dataspace Connector, I still have an issue trying to have two Dataspace Connectors talk to each other. Help on this would be appreciated very much. Here's what I did so far:
(1) I cloned the repository twice, in two directories on my VM running Ubuntu (latest LTS version).
(2) I amended the ports in one of the two connectors to localhost:8081 instead of the default localhost:8080 in two files: the `docker-compose.yml` and in `application.properties`.
(3) I follow the documentation and create data resource, representation, and actual data on one of the connectors (the 8080 one).
(4) I try to read the metadata of the 8080 connector from the 8081 connector using the `/admin/api/request/description` through the Swagger UI, passing `https://localhost:8080/api/ids/data` as the recipient.
(5) Subsequently, I receive an error:

`{
  "timestamp": "2020-12-09T10:12:32.659+0000",
  "status": 500,
  "error": "Internal Server Error",
  "message": "proxy.dortmund.isst.fraunhofer.de",
  "path": "/admin/api/request/description"
}`

Both connectors are running in Docker, I followed your recommendations regarding proxy settings, meaning I deleted them altogether in `config.json` after I got the same error using the default proxy settings. Do you have any hints on where I went wrong there? Thank you in advance!

**Comment by juliapampus** [comment, marked as answer]
Hi Vincent, 

I could not reproduce your problem so far. Are you using the current dev branch (updated yesterday) or some version before? We changed the error handling, so I may get a different response than you do. Furthermore, did you remove the Postgres from the `docker-compose.yml` or not? I guess you should start running two connectors in Docker by using one `docker-compose.yml`, not two. We will deal with a complete setup with two connectors and two external databases as soon as possible.

For an easy start, I suggest you either start two connectors locally simply using Maven for building a .jar file and runnig it with Java. Or you take a look at the `docker-compose.yml` in the [docker-test setup](https://github.com/FraunhoferISST/DataspaceConnector/releases/tag/v3.2.1) and copy it to your project folder. Furthermore, you need to remove the proxy settings from the connector's `config.json` and remove the SSL settings from the `application.properties` as the provided certificate for localhost is known to cause some problems in a Docker setup. For IDS communication, you can then follow the description at https://github.com/FraunhoferISST/DataspaceConnector/tree/v3.2.1#docker-setup.

Please let me know if you wer able to get the setup running.

Kind regards,
Julia

**Reply by Vincent-UBT** [reply]
Hi Julia,

thank you very much for the support! I managed to get the setup running with your recommendation. In addition, I was using an older version from some weeks ago, which might explain the differing response.

Best Regards
Vincent
